hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1368,8 +1385,24 @@ public List<BlockGroup> getPendingDeletionKeys(final int keyCount)
       while (keyIter.hasNext() && currentCount < keyCount) {
         KeyValue<String, RepeatedOmKeyInfo> kv = keyIter.next();
         if (kv != null) {
+          // Multiple keys with the same path can be queued in one DB entry
           RepeatedOmKeyInfo infoList = kv.getValue();
           for (OmKeyInfo info : infoList.cloneOmKeyInfoList()) {
+            // Skip the key if it exists in the previous snapshot (of the same
+            // scope) as in this case its blocks should not be reclaimed
+
+            // TODO: HDDS-7968
+            //  1. If previous snapshot keyTable has key info.getObjectID(),
+            //  skip it. Pending HDDS-7740 merge to reuse the util methods to
```

Comment:
cc: @aswinshakil  lets have a single place in the code to check for these conditions.

Reply:
The check in the previous snapshot should be objectID based check. Ha

Reply:
> The check in the previous snapshot should be objectID based check

Correct. Now that #4333 is merged we are one step closer towards handling that.